### PR TITLE
Raise minimum supported PostgreSQL version to 13

### LINKED ISSUE #1724: Raise minimum supported PostgreSQL version to 13
### Current Behavior

We currently support and test against PostgreSQL version 11. This version, as well as version 12, are [no longer supported](https://www.postgresql.org/support/versioning/) by the PostgreSQL project.

Image

### Proposed Behavior

Raise the lowest supported PostgreSQL version to 13, in alignment with the official support policy.

Update documentation, testcontainer versions, images in `docker-compose.yml` etc. accordingly.

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/hyades/blob/main/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/hyades/issues) for whether this enhancement was already requested

### PR DESCRIPTION
**Pull request by nscuro**
### Description

Raises minimum supported PostgreSQL version to 13.

Also ensures that tests are executed against this minimum supported version.

### Addressed Issue

Closes #1724

### Additional Details

N/A

### Checklist

- [x] I have read and understand the [contributing guidelines]
- ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- ~This PR implements an enhancement, and I have provided tests to verify that it works as intended~
- ~This PR introduces changes to the database model, and I have updated the [migration changelog] accordingly~
- [x] This PR introduces new or alters existing behavior, and I have updated the [documentation] accordingly

[contributing guidelines]: ../CONTRIBUTING.md#pull-requests
[documentation]: https://dependencytrack.github.io/hyades/latest/development/documentation/
[migration changelog]: https://dependencytrack.github.io/hyades/latest/development/database-migrations/

### CODE REVIEW

File: docs/operations/database.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ For example:
 ```yaml
 services:
   postgres:
-    image: postgres:16
+    image: postgres:17
```
Comment (sahibamittal) [inline docs/operations/database.md:94]: Shouldn't it be 13?

Comment (nscuro) [reply docs/operations/database.md:94]: In this case, the version is part of a snippet we show in the docs. Here I'd much rather promote users to pick a newer version of Postgres, rather than the minimum support one.

**Review by sahibamittal — APPROVED** [review]
